lib/CachedSource.js

```diff
@@ -131,48 +139,162 @@ class CachedSource extends Source {
 
 	sourceAndMap(options) {
 		const key = options ? JSON.stringify(options) : "{}";
-		let cacheEntry = this._cachedMaps.get(key);
-		if (cacheEntry && cacheEntry.map === undefined) {
+		const cacheEntry = this._cachedMaps.get(key);
```

Comment:
These changes here are mostly because I combined the various map representations into `_cachedMaps`, but since it required all these changes it may be desirable to store the different caches into different maps.

I can certainly go back and refactor this if desired, but I wanted to put this out here just to start a discussion and get some feedback on these changes before spending any more time on it.

Reply:
I refactored it a bit. The code started to get a bit confusing...